src/com.target.lorri.varlink

```diff
@@ -12,6 +12,38 @@ type ShellNix (
   path: string
 )
 
+# Monitor the daemon. The method will reply with an update whenever a build begins or ends.
+# Montior will immediately reply with a snapshot of known projects, then a marker event,
+# indicating that the stream of events is now "live."
+method Monitor() -> (event: Event)
+
+type Event (
+    kind: (sectionend, started, completed, failure),
+    nix_file: ?ShellNix, # not in sectionend
```

Comment:
```suggestion
    nix_file: ?ShellNix, # not present if kind == sectionend
```
That would be my suggestion - it's just a little more explicit. I would do something similar for the other comments.